kong/clustering/compat/removed_fields.lua

```diff
@@ -125,5 +125,27 @@ return {
     zipkin = {
       "propagation",
     },
+    ai_proxy = {
```

Comment:
similar to my comment [here](https://github.com/Kong/kong-ee/pull/9080#discussion_r1595618969), we should be able to remove logic from checkers now, that makes `response_streaming` and `model.options.upstream_path` nil [here](https://github.com/Kong/kong/blob/f02d4efc35ff821f66deed7a60e27577865a0e9b/kong/clustering/compat/checkers.lua#L33-L51)

Reply:
I'm scared to change stuff that's working right now.

Can we chore this for later?

Reply:
@gruceo

Reply:
ok, let's do it later

Reply:
IMHO: since [this is covered by tests](https://github.com/Kong/kong/blob/f02d4efc35ff821f66deed7a60e27577865a0e9b/spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua#L476), removing checkers logic will confirm things are working the right way (i.e. via the `removed_fields`) rather than through some other logic, so it would feel safer to me

Reply:
if we remove twice, it feels safer to me...

Reply:
I’m scared to remove this now that EE team will test and verify it. 

I can add to 3.7.1 rollup and make sure it’s still fixed by hand?